### PR TITLE
DOC-800 add enterprise properties to reference

### PR DESCRIPTION
**Pull request by Deflaimun**
## Description

Review deadline: Dec 11th

## Page previews

https://deploy-preview-904--redpanda-docs-preview.netlify.app/current/reference/properties/cluster-properties/
https://deploy-preview-904--redpanda-docs-preview.netlify.app/current/reference/properties/object-storage-properties/

CTRL+F for

audit_enabled
cloud_storage_enabled
core_balancing_continuous
default_leaders_preference
enable_schema_id_validation
http_authentication
iceberg_enabled (still the case in 24.3.1) 
partition_autobalancing_mode
sasl_mechanisms

## Checks

- [ ] New feature
- [x] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

File: modules/reference/pages/properties/cluster-properties.adoc [hunk 1]
```diff
@@ -1602,6 +1622,8 @@ Default value for the `redpanda.iceberg.delete` topic property that determines i
 
 === iceberg_enabled
 
+include::reference:partial$enterprise-licensed-property.adoc[]
```
Comment (Deflaimun) [inline modules/reference/pages/properties/cluster-properties.adoc:1625]: Iceberg being an enterprise property is still the case for 24.3.1

File: modules/reference/partials/enterprise-licensed-property.adoc [hunk 2]
```diff
@@ -0,0 +1 @@
+NOTE: This property requires a license when used with certain values.  Check the field *Enterprise license required*. For license details, see xref:get-started:licenses.adoc[].
```
Comment (Feediver1) [inline modules/reference/partials/enterprise-licensed-property.adoc]: my first question as a customer would be, what do you mean by "certain values"? The second sentence, "Check the field *Enterprise license required*."...Check it for what, exactly?

Comment (Deflaimun) [reply modules/reference/partials/enterprise-licensed-property.adoc]: take "sasl_mechanism" for example. Enterprise is not required when using `SCRAM` but it's required when using `GSSAPI` or `OAUTHBEARER`

Comment (Deflaimun) [reply modules/reference/partials/enterprise-licensed-property.adoc]: maybe ?
```suggestion
NOTE: Some values for this property require an Enterprise license. Refer to the *Enterprise license required* field for specific requirements. For license details, see xref:get-started:licenses.adoc[].
```

Comment (Feediver1) [reply modules/reference/partials/enterprise-licensed-property.adoc]: LGTM!

**Review by Feediver1 — APPROVED** [review]

**Review by michael-redpanda — APPROVED** [review]
This looks great!!!

**Pull request merged**